### PR TITLE
sql: add missing malloc failure handling

### PR DESCRIPTION
**Pull request by ImeevMA**
This patch fixes some possible bugs that may occur due to malloc
failure.

Closes https://github.com/tarantool/security/issues/65
Closes https://github.com/tarantool/security/issues/66
Closes https://github.com/tarantool/security/issues/68

### CODE REVIEW

File: src/box/sql/select.c [hunk 1]
```diff
@@ -740,8 +740,10 @@ addWhereTerm(Parse * pParse,	/* Parsing context */
 
 	struct Expr *pE1 = sql_expr_new_column(db, pSrc, iLeft, iColLeft);
 	struct Expr *pE2 = sql_expr_new_column(db, pSrc, iRight, iColRight);
-	if (pE1 == NULL || pE2 == NULL)
+	if (pE1 == NULL || pE2 == NULL) {
```
Comment (locker) [inline src/box/sql/select.c:743]: AFAIU if `pE1` allocation succeeds but `pE2` allocations fails, `pE1` will leak. This is hardly possible in reality, but static analyzers may still complain about it.

File: src/box/sql/select.c [hunk 2]
```diff
@@ -2188,22 +2192,25 @@ sqlColumnsFromExprList(Parse * parse, ExprList * expr_list,
 				if (zName[j] == '_')
 					nName = j;
 			}
-			zName =
-			    sqlMPrintf(db, "%.*z_%u", nName, zName, ++cnt);
+			zName = sqlMPrintf(db, "%.*z_%u", nName, zName, ++cnt);
```
Comment (locker) [inline src/box/sql/select.c:2195]: Just curious: doesn't it leak memory allocated for `zName` with `sqlDbStrDup` above?

File: src/box/sql/select.c [hunk 3]
```diff
@@ -2188,22 +2192,25 @@ sqlColumnsFromExprList(Parse * parse, ExprList * expr_list,
...
+		if (sqlHashInsert(&ht, zName, field) == field) {
+			sql_free(zName);
 			sqlOomFault(db);
-		space_def->fields[i].name = region_alloc(region, name_len + 1);
+			goto cleanup;
+		}
+		space_def->fields[i].name = region_alloc(region, nName + 1);
 		if (space_def->fields[i].name == NULL) {
 			sqlOomFault(db);
 			goto cleanup;
```
Comment (locker) [inline src/box/sql/select.c:2210]: `zName` isn't freed...

**Review by locker — CHANGES_REQUESTED** [review]
1. Missing references to the tickets you fix. Please add to both the PR and commits.
2. `NO_CHANGLOEG=bugfix` isn't a good excuse not to add a changlog. Let's instead write that we're unlikely to run into this bug in practice.
3. "sql: fix some possible errors" is too vague for a commit title. Let's change it so that it's clear what exactly we're trying to fix here ("sql: add missing malloc failure handling" maybe?)

File: src/box/sql/select.c (hunk 1 above)
Comment (ImeevMA) [reply src/box/sql/select.c]: True, fixed.

File: src/box/sql/select.c (hunk 2 above)
Comment (ImeevMA) [reply src/box/sql/select.c:2200]: No, these values are added to a hash table and removed during cleanup below.

File: src/box/sql/select.c (hunk 3 above)
Comment (ImeevMA) [reply src/box/sql/select.c:2215]: No, this value is added to a hash table and removed during cleanup below.

**Review by locker — APPROVED** [review]

**Pull request merged**